### PR TITLE
add guage panel

### PR DESCRIPTION
**Pull request by butlerx**
## What does this do?

Adds the ability to create guage panels in grafana 6 and up

## Why is it a good idea?

ability to take full advantage of grafana's built-in panels

### CODE REVIEW

**Comment by ljades** [comment]
Is there any update on if this or https://github.com/weaveworks/grafanalib/pull/179 are planning to be reviewed/merged?

**Comment by butlerx** [comment]
pr updated

File: grafanalib/core.py [hunk 1]
```diff
@@ -1577,3 +1593,144 @@ def to_json_data(self):
...
+    """Threshold for a gauge
+
+    :param color: color of threshold
+    :param index: index of color in gauge
+    :param value: when to use this color will be null if index is 0
+    """
+
+    color = attr.ib()
+    index = attr.ib(validator=instance_of(int))
+    value = attr.ib(validator=instance_of(int))
```
Comment (ljades) [inline grafanalib/core.py]: Value can be a float as well

File: grafanalib/core.py [hunk 2]
```diff
@@ -1577,3 +1593,144 @@ def to_json_data(self):
...
+    minSpan = attr.ib(default=None)
+    rangeMaps = attr.ib(default=attr.Factory(list))
+    repeat = attr.ib(default=None)
+    span = attr.ib(default=6)
+    thresholdLabels = attr.ib(default=False, validator=instance_of(bool))
+    thresholdMarkers = attr.ib(default=True, validator=instance_of(bool))
+    thresholds = attr.ib(
+        default=attr.Factory(
+            lambda: [
+                Threshold("green", 0),
```
Comment (ljades) [inline grafanalib/core.py]: Threshold `value` is currently a required positional argument, so this default breaks. Either give `value` a default value, or provide a `value` here.

**Comment by butlerx** [comment]
failing tests seem to be related to be realated to `attr` depricating convert keyword

File: CHANGELOG.rst [hunk 3]
```diff
@@ -8,11 +8,11 @@ Next release
 Changes
 -------
 
+* Add ``GuagePanel`` for creating guages in grafana 6
 * TBA
 
 
 0.5.5 (2020-02-17)
-=======
```
Comment (dholbach) [inline CHANGELOG.rst]: @matthewmrichter ^ it'd be good to re-add this.

**Pull request merged**